Event/AuthorizationRequestResolveEvent.php

```diff
@@ -9,49 +9,36 @@
 use LogicException;
 use Psr\Http\Message\ResponseInterface;
 use Symfony\Component\EventDispatcher\Event;
+use Trikoder\Bundle\OAuth2Bundle\League\Entity\User;
 
+/**
+ * Class AuthorizationRequestResolveEvent
+
+ * @package Trikoder\Bundle\OAuth2Bundle\Event
+ */
 final class AuthorizationRequestResolveEvent extends Event
 {
-    public const AUTHORIZATION_APPROVED = true;
-    public const AUTHORIZATION_DENIED = false;
-
     /**
      * @var AuthorizationRequest
      */
     private $authorizationRequest;
 
     /**
-     * @var ?ResponseInterface
+     * @var null|ResponseInterface
      */
     private $response;
 
-    /**
-     * @var bool
-     */
-    private $authorizationResolution = self::AUTHORIZATION_DENIED;
-
     public function __construct(AuthorizationRequest $authorizationRequest)
     {
         $this->authorizationRequest = $authorizationRequest;
     }
 
-    public function getAuhorizationResolution(): bool
-    {
-        return $this->authorizationResolution;
-    }
-
-    public function resolveAuthorization(bool $authorizationResolution): void
-    {
-        $this->authorizationResolution = $authorizationResolution;
-        $this->response = null;
```

Comment:
Here, if a listener approves or denies the authorization request, the response is unset.